src/solvers/sat/satcheck_cadical.cpp

```diff
@@ -79,6 +79,21 @@ propt::resultt satcheck_cadicalt::do_prop_solve()
   }
   else
   {
+    // if assumptions contains false, we need this to be UNSAT
```

Comment:
:pick: I don't understand this change or why the assume falses need special casing, but it should probably go in its own commit explaining in more detail why this is required

Reply:
Preferably with a test that only passes with this

Reply:
Just for the record: this is done in keeping with other solvers, and thus about the way in which we use the solver interface. I agree that a test would be useful.

Reply:
BTW, the reason why 'assume false' needs special casing is that SAT solvers don't have a 'false' literal.